_posts/2016-05-01-cors.markdown

```diff
@@ -0,0 +1,43 @@
+---
+layout: post
+title: "Implement CORS in Spark"
+author: <a href="https://www.linkedin.com/in/franklin-gauer-iii-4271666" target="_blank">Franklin M. Gauer III</a>
+date: 2016-05-01 22:22:22
+comments: true
+summary: >
+ Some boilerplate code to enable CORS on your Spark server.
+---
+## Enable CORS
+A little boilerplate code to enable CORS (Cross-Origin Resource Sharing) on your Spark server. This code opens CORS wide open on your server so please be cautious and modify the header values to suit your specific needs.
+
+<pre><code class="language-markup">
+// Enables CORS on requests. This method is an initialization method and should be called once.
+private static void enableCORS(final String origin, final String methods, final String headers) {
+
+  options("/*", (request,response)->{
+
+      String accessControlRequestHeaders = request.headers("Access-Control-Request-Headers");
+      if (accessControlRequestHeaders != null) {
+          response.header("Access-Control-Allow-Headers", accessControlRequestHeaders);
+      }
+
+      String accessControlRequestMethod = request.headers("Access-Control-Request-Method");
+      if(accessControlRequestMethod != null){
+        response.header("Access-Control-Allow-Methods", accessControlRequestMethod);
+      }
+
+      return "OK";
+  });		
+  
+    before(new Filter() {
```

Comment:
You use a lambda for the route in ```options()```, so you should probably use a lambda here too.